tt_metal/impl/buffers/global_semaphore.hpp

```diff
@@ -0,0 +1,42 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cstdint>
+#include <memory>
+
+#include "tt_metal/common/core_coord.hpp"
+#include "tt_metal/impl/buffers/buffer_constants.hpp"
+#include "tt_metal/llrt/hal.hpp"
+
+namespace tt::tt_metal {
+
+inline namespace v0 {
+class Buffer;
+class Device;
+}  // namespace v0
+
+class GlobalSemaphore {
+   public:
+    GlobalSemaphore(Device *device, CoreRangeSet cores, uint32_t initial_value, BufferType buffer_type = BufferType::L1);
```

Comment:
@patrickroberts / @tt-aho ,  should we make this GlobalSemaphore opaque?

Reply:
It's fine to leave this as-is, since you're not trying to put this in the `v1` namespace.